src/api-binder/report.ts

```diff
@@ -141,21 +144,45 @@ export async function startReporting() {
 	])) as StateReportOpts;
 
 	let reportPending = false;
-	const doReport = async () => {
+	// Reports current state if not already sending and does not exceed report
+	// frequency. Returns true if sent; otherwise false.
+	const doReport = async (): boolean => {
 		if (!reportPending) {
-			reportPending = true;
-			await reportCurrentState(reportConfigs);
-			reportPending = false;
+			const sinceLastReport = performance.now() - lastReportTime;
```

Comment:
Since `lastReportTime` is only used inside the `startReporting` function, I would declare it inside the function instead of as a global

Reply:
`reportCurrentState()` handles capturing and sending currentState, and this notion of when/what we actually capture is really import for the integrity of the report. So I would keep the logic in that function with definition of `lastReport`. `doReport()` is more of a user of `reportCurrentState()`.

Reply:
Ok, sounds good